src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/AbstractRenameStrategy.cs

```diff
@@ -63,8 +69,14 @@ protected bool HasMatchingSyntaxNode(Document document, SyntaxNode syntaxNode, s
                 kind == DeclarationKind.Enum ||
                 kind == DeclarationKind.Struct)
             {
-                return generator.GetName(syntaxNode) == name;
+                return string.Compare(generator.GetName(syntaxNode), name, !isCaseSensitive) == 0;
             }
+
+            if (_roslynServices.IsModuleDeclaration(syntaxNode))
```

Comment:
So it looks like all this plumbing to add a SyntaxFactsService is to know if this is a module. Can't you get that from the symbol?

var model = document.GetSemanticModelAsync();
var symbol = model.GetSymbolInfo(syntaxNode).Symbol as INamedTypeSymbol;

if (symbol.TypeKind == TypeKind.Module) ...